bianca_pipes.rb

```diff
@@ -0,0 +1,202 @@
+  @planets[:name] = planet_name
+  @planets[:mass] = mass
+  @planets[:orbit] = orbit
+  @planets[:tilt] = tilt
+  @planets[:rotation] = rotation
+  @planets[:num_of_moons] = num_of_moons
+end
+
+def add_hash(planet_name, mass, orbit, tilt, rotation, num_of_moons)
+  @planets[:name] = planet_name
+  @planets[:mass] = mass
+  @planets[:orbit] = orbit
+  @planets[:tilt] = tilt
+  @planets[:rotation] = rotation
+  @planets[:num_of_moons] = num_of_moons
+end
+
+#return list of planets as string
+def return_planets
+  planet_hash = ""
+  @planets.each do |key, value|
+    planet_hash += "#{key}: #{value}\n"
+  end
+  puts planet_hash
+end
+end
+
+milky_way = SolarSystem.new("milkyway", "Earth", 34, 5, 6, 7, 1)
+
+milky_way.return_planets
+
+####Wave_2 & Wave_3 #####
+
+class SolarSystem
+
+  #use @planets instance variable
+  def initialize(planets)
+    @planets = Array.new(planets)
+  end
+
+  #method to add planet to list
+  def add(planet)
+    @planet = planet
+    @planets << planet
+    return @planet
+  end
+
+  #create interface for user to interact with solar system
+  #can select planet and view info about it
+
+  def interact
+    puts "What planet would you like to view and learn about?"
+    planet = gets.chomp
+    @planets.each do |series_of_planets|
+      if series_of_planets == planet
+        puts "#{planet.name}"
+        puts "#{planet.weight}"
+        puts "#{planet.size}"
+        puts "#{planet.orbit}"
+      else
+        puts "Sorry, it appears that the planet you want is not available."
+      end
+    end
+  end
+
+  #method to return, not print list of planets as String
+  def list
+    num = 0
+    planet_list = ""
+    @planets.each do |planet|
+      num += 1
+      planet_list += "#{num}. #{planet.name}\n"
+    end
+    return planet_list
+  end
+
+  def planet_list_class
+    full_list_planets = ""
+    @planets.each do |full_list|
+      full_list_planets += full_list.print_list
+    end
+    return full_list_planets
+  end
+
+  def view_planet
+    puts "What planet would you like to learn about?"
+    view_planet = gets.chomp
+
+    @planets.each do |planet|
+      name = "#{planet.name}"
+      if view_planet == name
+        puts "You would like to learn more about #{name}."
+        puts "Name: #{planet.name}"
+        puts "Weight: #{planet.weight}"
+        puts "Size: #{planet.size}"
+        puts "Orbit: #{planet.orbit}"
+      end
+    end
+  end
+
+  def add_planet
+      puts "You have selected to add a planet."
+      puts "Please type in the name of the planet."
+      name = gets.chomp
+      puts "Please type in the weight."
+      weight = gets.chomp.to_i
+      puts "Please type in the size."
+      size = gets.chomp.to_i
+      puts "Please type in the orbit."
+      orbit = gets.chomp.to_i
+      planet_array = Planet.new(name, weight, size, orbit)
+      @planets <<  planet_array
```

Comment:
Here, what you've labeled as `planet_array` is not an array but a single `Planet`.